sources/metricdog/src/metricdog.rs

```diff
@@ -105,8 +111,29 @@ impl Metricdog {
 
     /// Sends a notification to the metrics url that boot succeeded.
     pub(crate) fn send_boot_success(&self) -> Result<()> {
-        // timeout of 3 seconds to prevent blocking the completion of mark-boot-success
-        self.send("metricdog", "boot_success", None, Some(3))?;
+        let mut values = HashMap::new();
+        // Note whether this is the first boot or not
+        values.insert(
+            "is_first_boot".to_string(),
+            self.host_check.is_first_boot()?.to_string(),
+        );
+        // Record how long it took for each stage of the boot process and send it as part of the
+        // boot success event. The time is recorded in microseconds.
+        values.insert(
+            "preconfigured_time".to_string(),
+            self.host_check.preconfigured_time()?,
+        );
+        values.insert(
+            "configured_time".to_string(),
+            self.host_check.configured_time()?,
+        );
+        values.insert(
+            "network_ready_time".to_string(),
+            self.host_check.network_ready_time()?,
+        );
```

Comment:
If we want these values to be stable, to simplify querying the results - would it make sense to have a real struct with known, optional fields? Or an enum with two variants - `BootStatus` and `HealthStatus`?